File: tools/taos-tools/deps/avro/lang/java/avro/src/main/java/org/apache/avro/file/DataFileConstants.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.avro.file;

/**
 * Constants used in data files.
 */
public class DataFileConstants {
  private DataFileConstants() {
  } // no public ctor

  public static final byte VERSION = 1;
  public static final byte[] MAGIC = new byte[] { (byte) 'O', (byte) 'b', (byte) 'j', VERSION };
  public static final long FOOTER_BLOCK = -1;
  public static final int SYNC_SIZE = 16;
  public static final int DEFAULT_SYNC_INTERVAL = 4000 * SYNC_SIZE;

  public static final String SCHEMA = "avro.schema";
  public static final String CODEC = "avro.codec";
  public static final String NULL_CODEC = "null";
  public static final String DEFLATE_CODEC = "deflate";
  public static final String SNAPPY_CODEC = "snappy";
  public static final String BZIP2_CODEC = "bzip2";
  public static final String XZ_CODEC = "xz";
  public static final String ZSTANDARD_CODEC = "zstandard";

}

File: tools/taos-tools/deps/avro/lang/csharp/src/apache/main/File/DataFileConstants.cs
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

namespace Avro.File
{
    /// <summary>
    /// Constants used in data files.
    /// </summary>
    [System.Diagnostics.CodeAnalysis.SuppressMessage("Design",
        "CA1052:Static holder types should be Static or NotInheritable",
        Justification = "Maintain public API")]
    public class DataFileConstants
    {
        /// <summary>
        /// Key for the 'sync' metadata entry.
        /// </summary>
        public const string MetaDataSync = "avro.sync";

        /// <summary>
        /// Key for the 'codec' metadata entry.
        /// </summary>
        public const string MetaDataCodec = "avro.codec";

        /// <summary>
        /// Key for the 'schema' metadata entry.
        /// </summary>
        public const string MetaDataSchema = "avro.schema";

        /// <summary>
        /// Identifier for the null codec.
        /// </summary>
        public const string NullCodec = "null";

        /// <summary>
        /// Identifier for the deflate codec.
        /// </summary>
        public const string DeflateCodec = "deflate";

        /// <summary>
        /// Reserved 'avro' metadata key.
        /// </summary>
        public const string MetaDataReserved = "avro";

        /// <summary>
        /// Avro specification version.
        /// </summary>
        public const int Version = 1;

        /// <summary>
        /// Magic bytes at the beginning of an Avro data file.
        /// </summary>
        public static byte[] Magic = { (byte)'O',
                                       (byte)'b',
                                       (byte)'j',
                                       Version };

        /// <summary>
        /// Hash code for the null codec.
        /// </summary>
        /// <seealso cref="NullCodec.GetHashCode()"/>
        public const int NullCodecHash = 2;

        /// <summary>
        /// Hash code for the deflate codec.
        /// </summary>
        /// <seealso cref="DeflateCodec.GetHashCode()"/>
        public const int DeflateCodecHash = 0;

        /// <summary>
        /// Size of a sync token in bytes.
        /// </summary>
        public const int SyncSize = 16;

        /// <summary>
        /// Default interval for sync tokens.
        /// </summary>
        public const int DefaultSyncInterval = 4000 * SyncSize;
    }
}
